File: wrappers/dotnet/indy-sdk-dotnet-test/LedgerTests/PoolConfigRequestTest.cs
using Hyperledger.Indy.DidApi;
using Hyperledger.Indy.LedgerApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.LedgerTests
{
    [TestClass]
    public class PoolConfigRequestTest : IndyIntegrationTestWithPoolAndSingleWallet
    {
        [TestMethod]
        public async Task TestBuildPoolConfigRequestWorks()
        {
            var expectedResult = string.Format("\"identifier\":\"{0}\"," +
                "\"operation\":{{" +
                "\"type\":\"111\"," +
                "\"writes\":false," +
                "\"force\":false" +
                "}}", DID);

            var request = await Ledger.BuildPoolConfigRequestAsync(DID, false, false);

            Assert.IsTrue(request.Contains(expectedResult));
        }

        [TestMethod]
        public async Task TestPoolConfigRequestWorks()
        {
            var didResult = await Did.CreateAndStoreMyDidAsync(wallet, TRUSTEE_IDENTITY_JSON);
            var did = didResult.Did;

            var request = await Ledger.BuildPoolConfigRequestAsync(did, false, false);
            await Ledger.SignAndSubmitRequestAsync(pool, wallet, did, request);

            request = await Ledger.BuildPoolConfigRequestAsync(did, true, false);
            await Ledger.SignAndSubmitRequestAsync(pool, wallet, did, request);
        }
    }
}

File: wrappers/android/src/androidTest/java/org/hyperledger/indy/sdk/ledger/PoolConfigRequestTest.java
package org.hyperledger.indy.sdk.ledger;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithPoolAndSingleWallet;
import org.hyperledger.indy.sdk.did.Did;
import org.hyperledger.indy.sdk.did.DidResults;
import org.junit.Test;

import static org.junit.Assert.*;

public class PoolConfigRequestTest extends IndyIntegrationTestWithPoolAndSingleWallet {

	@Test
	public void testBuildPoolConfigRequestWorks() throws Exception {
		String expectedResult = String.format("\"identifier\":\"%s\"," +
				"\"operation\":{" +
				"\"type\":\"111\"," +
				"\"writes\":false," +
				"\"force\":false" +
				"}", DID);

		String request = Ledger.buildPoolConfigRequest(DID, false, false).get();

		assertTrue(request.contains(expectedResult));
	}

	@Test
	public void testPoolConfigRequestWorks() throws Exception {
		DidResults.CreateAndStoreMyDidResult didResult = Did.createAndStoreMyDid(wallet, TRUSTEE_IDENTITY_JSON).get();
		String did = didResult.getDid();

		String request = Ledger.buildPoolConfigRequest(did, false, false).get();
		Ledger.signAndSubmitRequest(pool, wallet, did, request).get();

		request = Ledger.buildPoolConfigRequest(did, true, false).get();
		Ledger.signAndSubmitRequest(pool, wallet, did, request).get();
	}
}
